presto-parquet/src/test/java/io/prestosql/parquet/TestTupleDomainParquetPredicate.java

```diff
@@ -315,6 +322,55 @@ public void testDate()
                 .withMessage("Corrupted statistics for column \"DateColumn\" in Parquet file \"testFile\": [min: 200, max: 100, num_nulls: 0]");
     }
 
+    @Test
+    public void testTimestampMilli()
+            throws ParquetCorruptionException
+    {
+        Instant now = Instant.now();
+        String column = "timestampColumn";
+        ShortTimestampType timestampType = new ShortTimestampType(3);
+        assertEquals(getDomain(timestampType, 0, null, ID, column, true), all(timestampType));
+        assertEquals(getDomain(timestampType, 10, timestampColumnStats(now, now), ID, column, true), singleValue(timestampType, now.toEpochMilli()));
+        assertEquals(
+                getDomain(timestampType, 10, timestampColumnStats(now.minusSeconds(10), now), ID, column, true),
+                create(ValueSet.ofRanges(range(timestampType, now.minusSeconds(10).toEpochMilli(), true, now.toEpochMilli(), true)), false));
+
+        // ignore corrupted statistics
+        assertEquals(getDomain(timestampType, 10, timestampColumnStats(now.plusSeconds(10), now), ID, column, false), create(ValueSet.all(timestampType), false));
+        // fail on corrupted statistics
+        assertThatExceptionOfType(ParquetCorruptionException.class)
+                .isThrownBy(() -> getDomain(timestampType, 10, timestampColumnStats(now.plusSeconds(10), now), ID, column, true))
+                .withMessageMatching("Corrupted statistics for column \"timestampColumn\" in Parquet file \"testFile\":.*");
+    }
+
+    private static BinaryStatistics timestampColumnStats(Instant minimum, Instant maximum)
+    {
+        BinaryStatistics statistics = new BinaryStatistics();
+        statistics.setMinMax(Binary.fromConstantByteArray(toParquetEncoding(minimum)), Binary.fromConstantByteArray(toParquetEncoding(maximum)));
+        return statistics;
+    }
+
+    private static byte[] toParquetEncoding(Instant timestamp)
+    {
+        long startOfDay = LocalDate.ofInstant(timestamp, ZoneOffset.UTC).atStartOfDay().toInstant(ZoneOffset.UTC).toEpochMilli();
+        long timeOfDayNanos = (long) ((timestamp.toEpochMilli() - startOfDay) * Math.pow(10, 6));
+
+        ByteBuffer encoding = ByteBuffer.allocate(12).order(ByteOrder.LITTLE_ENDIAN);
+        encoding.putLong(timeOfDayNanos);
+        encoding.putInt(millisToJulianDay(timestamp.toEpochMilli()));
+        byte[] arr = new byte[12];
+        encoding.position(0).get(arr, 0, 12);
+        return arr;
+    }
+
+    private static final int JULIAN_EPOCH_OFFSET_DAYS = 2_440_588;
+    private static final long MILLIS_IN_DAY = TimeUnit.DAYS.toMillis(1);
+
+    private static int millisToJulianDay(long timestamp)
+    {
+        return (int) (timestamp / MILLIS_IN_DAY) + JULIAN_EPOCH_OFFSET_DAYS;
```

Comment:
- we usually just `[TimeUnit.]MILLISECONDS.toDays(timestamp)`
- avoid unckeched `(int)` cast

```suggestion
        return toIntExact(MILLISECONDS.toDays(timestamp) + JULIAN_EPOCH_OFFSET_DAYS);
```